docs/contributing/provider-design.md

```diff
@@ -0,0 +1,129 @@
+
+## Resource Type Considerations
+
+Terraform resources work best as the smallest infrastructure blocks on which practitioners can build more complex configurations and abstractions, such as [Terraform Modules](https://www.terraform.io/docs/modules/). The general heuristic guiding when to implement a new Terraform resource for an aspect of AWS is whether the AWS service API provides create, read, update, and delete (CRUD) operations. However, not all AWS service API functionality falls cleanly into CRUD lifecycle management. In these situations, there is extra consideration necessary for properly mapping API operations to Terraform resources.
+
+This section highlights design patterns when to consider an implementation within a singular Terraform resource or as separate Terraform resources.
+
+Please note: the overall design and implementation across all AWS functionality is federated: individual services may implement concepts and use terminology differently. As such, this guide is not exhaustive. The aim is to provide general concepts and basic terminology that points contributors in the right direction, especially in understanding previous implementations.
+
+### Authorization and Acceptance Resources
+
+Some AWS services use an authorization-acceptance model for cross-account associations or access. Examples include:
+
+* Direct Connect Association Proposals
+* GuardDuty Member Invitations
+* RAM Resource Share Associations
+* Route 53 VPC Associations
+* Security Hub Member Invitations
+
+Depending on the API and components, AWS uses two basic ways of creating cross-region and cross-account associations. One way is to generate an invitation (or proposal) identifier from one AWS account to another. Then in the other AWS account, that identifier is used to accept the invitation. The second way is configuring a reference to another AWS account identifier. These may not require explicit acceptance on the receiving account to finish creating the association or begin working.
+
+To model creating an association using an invitation or proposal, follow these guidelines.
+
+* Follow the naming in the AWS service API to determine whether to use the term "invitation" or "proposal."
+* For the originating account, create an "invitation" or "proposal" resource. Make sure that the AWS service API has operations for creating and reading invitations. 
+* For the responding account, create an "accepter" resource. Ensure that the API has operations for accepting, reading, and rejecting invitations in the responding account. Map the operations as follows:
+	* Create: Accepts the invitation.
+	* Read: Reads the invitation to determine its status. Note that in some APIs, invitations expire and disappear, complicating associations. If a resource does not find an invitation, the developer should implement a fall back to read the API resource associated with the invitation/proposal.
+	* Delete: Rejects or otherwise deletes the invitation.
```

Comment:
```suggestion
    * Create: Accepts the invitation.
    * Read: Reads the invitation to determine its status. Note that in some APIs, invitations expire and disappear, complicating associations. If a resource does not find an invitation, the developer should implement a fall back to read the API resource associated with the invitation/proposal.
    * Delete: Rejects or otherwise deletes the invitation.
```